src/main/java/org/embl/mobie/viewer/MoBIE.java

```diff
@@ -371,23 +372,29 @@ public synchronized ImageSource getSource( String sourceName )
 	public SourceAndConverter< ? > openSourceAndConverter( String sourceName, String log )
 	{
 		final ImageSource imageSource = getSource( sourceName );
-		final String imagePath = getImagePath( imageSource );
-
-		if( log != null )
-			IJ.log( log + imagePath );
-
-		final ImageDataFormat imageDataFormat = settings.values.getImageDataFormat();
+        Set<ImageDataFormat> tmp = imageSource.imageData.keySet();
+        ImageDataFormat imageDataFormat = null;
+        for ( ImageDataFormat f : tmp ) {
+            if ( settings.values.getImageDataFormat().contains( f ) ) {
+                imageDataFormat = f;
+            }
+        }
+        if (imageDataFormat == null) {
+            System.err.println( "Error opening: " + imageSource );
+            throw new RuntimeException();
+        }
 
+		final String imagePath = getImagePath( imageSource, imageDataFormat );
+        if( log != null )
+            IJ.log( log + imagePath );
+		IJ.log( "Opening image:\n" + imagePath );
 		try
 		{
 			SpimData spimData = tryOpenSpimData( imagePath, imageDataFormat );
 			sourceNameToImgLoader.put( sourceName, spimData.getSequenceDescription().getImgLoader() );
 
 			final SourceAndConverterFromSpimDataCreator creator = new SourceAndConverterFromSpimDataCreator( spimData );
 			SourceAndConverter< ? > sourceAndConverter = creator.getSetupIdToSourceAndConverter().values().iterator().next();
-			// Touch the source once to initiate the cache,
-			// as this speeds up future accesses significantly
-			sourceAndConverter.getSpimSource().getSource( 0,0 );
```

Comment:
@KateMoreva Why did you remove this? This is important for the performance.

Reply:
Oh, it is probably merge mistake.

Reply:
@tischi, should be fixed